### ISSUE TITLE
[SCIF] appnames with "-" environment export

### ISSUE THREAD

**Issue by vsoch** [apptainer/singularity#1237]
There is a bug that a SCIF app that has a dash in it ("-") will not export some of its environment characters correctly. I'm fixing this now! It makes sense that it was done, because if you try to export a name with a "-" in it, you get:

```
singularity exec --app hub sregistry.simg ls /scif/apps/google-storage/scif
/.singularity.d/actions/exec: 1: /.singularity.d/env/94-appsbase.sh: SCIF_APPDATA_google-drive=/scif/data/google-drive: not found
```
and so it's reasonable to maintain the standard that app names can include these slashes (I like them in fact) but the environment variables do a general replacement of slashes with underscore.

**Comment by Saford91** [comment]
Hello @vsoch , I've seen this behavior with periods in SCIF app names. Can you translate those as well?

**Comment by vsoch** [comment]
Sure thing, could you open an issue here --> https://github.com/vsoch/scif then I'll test there and do the change here? The Scientific Filesystem is a standalone thing and Singularity is the first client with native integration, so I think it would be good to get it flushed out there first before fixing here (it's much harder to get PRs merged these days so we want to get it the first time!). Thank you!

**Comment by vsoch** [comment]
Nevermind I can do it for you!

**Comment by Saford91** [comment]
Too late!

**Comment by vsoch** [comment]
I know you are so fast!! This is me --> :turtle:

**Comment by vsoch** [comment]
see https://github.com/singularityware/singularity/pull/1272 @Saford91 !

**Comment by Saford91** [comment]
Thanks, @vsoch !

**Comment by vsoch** [comment]
No problemo! I think as soon as the other maintainers are able to review it should be very trivial to merge away. Glad to help.

**Comment by dtrudg** [comment]
Closing - as a fix for this was merged as #1375. Please let us know if any further issues.
